src/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -598,5 +594,86 @@ public void Dispose()
             _buildEvents.OnBuildBegin -= BuildEvents_OnBuildBegin;
             _solutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
         }
+
+        #region ILogger implementation
+        public void LogDebug(string data)
+        {
+            LogToVS(VerbosityLevel.Diagnostic, data);
+        }
+
+        public void LogVerbose(string data)
+        {
+            LogToVS(VerbosityLevel.Detailed, data);
+        }
+
+        public void LogInformation(string data)
+        {
+            LogToVS(VerbosityLevel.Normal, data);
+        }
+
+        public void LogWarning(string data)
+        {
+            LogToVS(VerbosityLevel.Minimal, data);
+        }
+
+        public void LogError(string data)
+        {
+            LogToVS(VerbosityLevel.Quiet, data);
+        }
+        #endregion ILogger implementation
+
+        private void LogToVS(VerbosityLevel verbosityLevel, string message)
+        {
+            if (Token.IsCancellationRequested)
```

Comment:
add a comment explaining why you are checking for cancellation inside a logger (it is very weird), also setting the canceled flag in here is odd and will not stand when you pull the logger out